`libvirt` fails to compile on `aarch64-darwin`

### Describe the bug

`$ nix-build -A nixopsUnstable --argstr system aarch64-darwin`

```
       last 10 log lines:
       > clang-11: warning: argument unused during compilation: '-L/nix/store/m57j19s9bs1jsfc0r9cbmgxkgyil0j1r-clang-11.1.0-lib/lib' [-Wunused-command-line-argument]
       > clang-11: warning: argument unused during compilation: '-L/nix/store/2ja4hxp27mp2h4cx567az88978mh3n10-libcxx-11.1.0/lib' [-Wunused-command-line-argument]
       > [556/1084] Linking target src/libvirt.0.dylibremote_remote_daemon_dispatch.c.o.c.o[Kackend_fs.c.o
       > FAILED: src/libvirt.0.dylib
       > clang  -o src/libvirt.0.dylib  -Wl,-dead_strip_dylibs -Wl,-headerpad_max_install_names -Wl,-undefined,error -shared -install_name @rpath/libvirt.0.dylib -compatibility_version 0 -current_version 0 -Wl,-force_load src/util/libvirt_util.a -Wl,-force_load src/conf/libvirt_conf.a -Wl,-force_load src/rpc/libvirt_net_rpc.a -Wl,-force_load src/rpc/libvirt_net_rpc_client.a -Wl,-force_load src/rpc/libvirt_net_rpc_server.a -Wl,-force_load src/access/libvirt_access.a -Wl,-force_load src/cpu/libvirt_cpu.a -Wl,-force_load src/hypervisor/libvirt_hypervisor.a -Wl,-force_load src/vmx/libvirt_vmx.a -Wl,-force_load src/remote/libvirt_remote_driver.a -Wl,-force_load src/locking/libvirt_lock_driver.a -Wl,-force_load src/logging/libvirt_log_driver.a -Wl,-force_load src/security/libvirt_security_manager.a -Wl,-force_load src/storage_file/libvirt_storage_file.a -Wl,-force_load src/esx/libvirt_driver_esx.a -Wl,-force_load src/test/libvirt_test.a -Wl,-force_load src/vmware/libvirt_vmware.a -Wl,-force_load src/libvirt_driver.a -Wl,-flat_namespace -Wl,-M,/private/tmp/nix-build-libvirt-7.9.0.drv-0/libvirt-7.9.0/build/src/libvirt.syms -Wl,-export_dynamic /nix/store/4spbly7vsc7d5xd03851c59833g7psv5-glib-2.70.0/lib/libglib-2.0.dylib /nix/store/4spbly7vsc7d5xd03851c59833g7psv5-glib-2.70.0/lib/libgobject-2.0.dylib /nix/store/4spbly7vsc7d5xd03851c59833g7psv5-glib-2.70.0/lib/libgio-2.0.dylib /nix/store/z4vigwb9q991yhxvlhmp6xjk7vkc1f8g-libxml2-2.9.12/lib/libxml2.dylib /nix/store/yy5qk7s3c5f9zzby4lsfdnkswr59xgbg-gnutls-3.7.2/lib/libgnutls.dylib -lm -lutil -Wl,-export_dynamic /nix/store/89rvdv89pi9wdscnxl4nj8zbkggxd0m7-yajl-2.1.0/lib/libyajl.dylib -Wl,-export_dynamic /nix/store/9g1am1kmhpwdah1b5yc8dwmi03rrfdc4-libssh2-1.10.0/lib/libssh2.dylib -Wl,-export_dynamic -lc -Wl,-export_dynamic -Wl,-export_dynamic -Wl,-export_dynamic -Wl,-export_dynamic -Wl,-export_dynamic -Wl,-export_dynamic -Wl,-export_dynamic -Wl,-export_dynamic -Wl,-export_dynamic -Wl,-export_dynamic -Wl,-export_dynamic /nix/store/glfqysk47db91b4zasf28ng770b9m71x-curl-7.76.1/lib/libcurl.dylib -Wl,-export_dynamic -Wl,-export_dynamic -Wl,-export_dynamic -ldl -Wl,-export_dynamic
       > ld: warning: ignoring file /private/tmp/nix-build-libvirt-7.9.0.drv-0/libvirt-7.9.0/build/src/libvirt.syms, building for macOS-arm64 but attempting to link with file built for unknown-unsupported file format ( 0x23 0x20 0x57 0x41 0x52 0x4E 0x49 0x4E 0x47 0x3A 0x20 0x67 0x65 0x6E 0x65 0x72 )
       > ld: file not found: /System/Library/Frameworks/Carbon.framework/Versions/A/Carbon for architecture arm64
       > clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
       > [563/1084] Compiling C object src/virtsecretd.p/remote_remote_daemon_dispatch.c.o
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/0fsf75fxs1xr5ssg5rfj2jhnyxgxvzcn-libvirt-7.9.0.drv'.
```


### Additional context

Found in #146481

It does build on `aarch64-linux` and `x86_64-darwin`.

Refs ZHF #144627

### Notify maintainers
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@fpletz @globin

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
 - libvirt
 - nixopsUnstable
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/development/libraries/libvirt/default.nix
+++ b/pkgs/development/libraries/libvirt/default.nix
@@ -60,6 +60,8 @@
 , ceph
 , enableGlusterfs ? false
 , glusterfs
+, Carbon
+, AppKit
 }:
 
 with lib;
@@ -152,6 +154,8 @@ stdenv.mkDerivation rec {
   ] ++ optionals stdenv.isDarwin [
     libiconv
     gmp
+    Carbon
+    AppKit
   ];
 
   preConfigure =

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -18356,7 +18356,9 @@ with pkgs;
 
   libversion = callPackage ../development/libraries/libversion { };
 
-  libvirt = callPackage ../development/libraries/libvirt { };
+  libvirt = callPackage ../development/libraries/libvirt {
+    inherit (darwin.apple_sdk.frameworks) Carbon AppKit;
+  };
   libvirt_5_9_0 = callPackage ../development/libraries/libvirt/5.9.0.nix { };
 
   libvirt-glib = callPackage ../development/libraries/libvirt-glib { };

